### PR TITLE
Use `getHeadingList` from `marked-gfm-heading-id` plugin

### PR DESCRIPTION
**Pull request by patrickpatrickpatrick**
## What

Delete `extractHeadingPageHeadings()`. Write new Metalsmith plugin that adds headings to each `file` object after each markdown file has been processed. 

Addresses issue https://github.com/alphagov/govuk-design-system/issues/3171.

## Why

Remove extra implementation to make use of already implemented feature of `marked-gfm-heading-id`. This method of implementation means we don't have parse the markdown multiple times and removes a loop of all the processed files.

### CODE REVIEW

File: lib/render-markdown.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+const { DesignSystemMarked } = require('./marked/index.js') // Markdown renderer
+const { slugify } = require('./nunjucks/filters')
+
+const marked = new DesignSystemMarked()
+/**
+ * Metalsmith markdown plugin
+ *
+ * Renders markdown and extracts headings from markdown
+ * and assigns to 'headings' variable of each file.
```
Comment (romaricpascal) [inline lib/render-markdown.js:12]: Cheers for adding a little description to the JSDoc on top of the title 😊

**Review by romaricpascal — COMMENTED** [review]
Cheers for picking this one up. The new plugin looks neat. 🙌🏻 

Given it does a little more than simple markdown rendering now and is responsible for extracting headings info as well, I'd be keen to keep the tests we had for `extract-page-headings` to validate that it does that part of the work correctly. 

The tests of the `metalsmith-lunr-index` will validate some of that, but having specific examples for each feature of the heading extraction (aliases, excluding from navigation, excluding from search) will make it easier to understand what exactly the heading extraction is about.

Would you mind adding them back (that'll probably require the plugin file to move in a folder in a similar way that the previous plugin was set up).

**Comment by patrickpatrickpatrick** [comment]
> Cheers for picking this one up. The new plugin looks neat. 🙌🏻
> 
> Given it does a little more than simple markdown rendering now and is responsible for extracting headings info as well, I'd be keen to keep the tests we had for `extract-page-headings` to validate that it does that part of the work correctly.
> 
> The tests of the `metalsmith-lunr-index` will validate some of that, but having specific examples for each feature of the heading extraction (aliases, excluding from navigation, excluding from search) will make it easier to understand what exactly the heading extraction is about.
> 
> Would you mind adding them back (that'll probably require the plugin file to move in a folder in a similar way that the previous plugin was set up), please?

Makes sense. Was able to just add them all back with minimal changes :)

**Review by romaricpascal — APPROVED** [review]
> Makes sense. Was able to just add them all back with minimal changes :)
🌟 Ace, hoped that we could reuse most of what was there. Thanks for updating.

**Pull request merged**